Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm

```diff
@@ -3061,4 +3038,36 @@ InstanceMethodSwizzler swizzler(PAL::getWTWritingToolsClass(), @selector(showPan
     TestWebKitAPI::Util::run(&finished);
 }
 
+TEST(WritingTools, CompositionWithOmittedTrailingWhitespaceContent)
+{
+    NSString *htmlString = @""
+    "<style>\n"
+    ".container {\n"
+    "   display:flex;\n"
+    "}\n"
+    "\n"
+    "img {\n"
+    "   display:block;\n"
+    "   width: 50px;\n"
+    "   height: 5px;\n"
+    "}\n"
+    "</style>\n"
+    "some<span class=container><span><img> </span>\n"
+    "</span>text\n"
+    "";
+
+    RetainPtr session = adoptNS([[WTSession alloc] initWithType:WTSessionTypeComposition textViewDelegate:nil]);
+    RetainPtr webView = adoptNS([[WritingToolsWKWebView alloc] initWithHTMLString:htmlString]);
+    [webView focusDocumentBodyAndSelectAll];
+
+    __block bool finished = false;
+    [[webView writingToolsDelegate] willBeginWritingToolsSession:session.get() requestContexts:^(NSArray<WTContext *> *contexts) {
+        EXPECT_EQ(1UL, contexts.count);
```

Comment:
Should we be asserting more things about the context? Curious how this test fails without your change, I thought we just give incorrect ranges.

Reply:
this test excersises the non-non-editable case, which already checks internally if the ranges differ, and if they do, asserts in debug mode, and returns an empty array of contexts, so this check is sufficient. Checking the length of the context itself isn't really necessary or strictly correct in validating this change, since what is important is that the ranges are equal, and not a particular range length.